src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicConnection.cs

```diff
@@ -74,7 +74,6 @@ internal sealed class State
             // Backlog limit is managed by MsQuic so it can be unbounded here.
             public readonly Channel<MsQuicStream> AcceptQueue = Channel.CreateUnbounded<MsQuicStream>(new UnboundedChannelOptions()
             {
-                SingleReader = true,
                 SingleWriter = true,
```

Comment:
Is SingleWriter correct still?

Reply:
I think so, but we should definitely double check this...

We should only be enqueuing to the channel from one thread -- the MsQuic callback thread.

But I'm not 100% sure we always call Complete/TryComplete from that thread.

I assume "SingleWriter" encompasses Complete/TryComplete as well, right?

Reply:
> I assume "SingleWriter" encompasses Complete/TryComplete as well, right?

Yup.  Doesn't need to be the exact same thread, just one at a time.

Reply:
Yes, I think it can be different thread but it is always one at the time as MsQuic would serialize events through queue. 
cc: @nibanks for final confirmation.

Reply:
MsQuic serializes all callbacks for the entire connection (and all child streams). It may change threads occasionally to align with the RSS core, but it never gives parallel callbacks.